text cell stuck in TeX mode #3857

kpaharelau opened the pull request:
only text block scanned by markdown

use for a testing
https://www.dropbox.com/s/i9of3t051hb5bd9/3857.bkr?dl=0

kpaharelau committed:
create test for category plot notebook #3667

fix test fail

kpaharelau committed:
create test for category plot notebook #3667

remove flaky checks

mnavasiolava committed:
flatten column menu #3672

mnavasiolava committed:
tables: add hysteresis to column filter #3673

EfimovVladimir committed:
add to help menu #3674

mnavasiolava committed:
remove unused code

kpaharelau committed:
Wrong chart size in Firefox browser #3683

use simple width() and height() functions in a firefox browser.

mnavasiolava committed:
#3673 hysteresis was removed from column search

mnavasiolava committed:
#3672 shortcuts for Data Bars and Heatmap are shown in menu items

kpaharelau committed:
Wrong chart size in Firefox browser #3683

these should be application-wide helper functions, not just here.
in fact i am surprised we don't have something like this already, please check.

I moved functionality to the GLOBALS. I did not find something like this. I searched by words "firefox", 'chrome', "browser"

kpaharelau committed:
Wrong chart size in Firefox browser #3683

should go in helper.js, global.js is just for constants not for functions.

also, please only define firefox, safari, and chrome as we support only them.

tomca32 committed:
Made session final in addSocketListener method of NamespaceService

scottdraves committed:
Merge pull request #3691 from twosigma/tom/fix-final

Made session final in addSocketListener method of NamespaceService

EfimovVladimir committed:
#36774 - fix test for help menu

kpaharelau committed:
Wrong chart size in Firefox browser #3683

Safari not Safary

mnavasiolava committed:
flatten column menu #3672. menu groups separators

mnavasiolava committed:
tables flash Processing... #3671. try to set timeout

tomca32 committed:
using beakerConfig instead of beakerConfigPref to get zombieLogging value

mnavasiolava committed:
wrong character was removed (debug fix)

scottdraves committed:
Merge pull request #3693 from twosigma/marina/remove-wrong-characters

Remove wrong character

mnavasiolava committed:
make table columns resizable by dragging #3582. truncate the content if column is narrower than the content in it.

mnavasiolava committed:
make table columns resizable by dragging #3582. works for fixed columns

kpaharelau committed:
colors differ between plot and legend #3669

property scope.legendDone has been corrected

mnavasiolava committed:
make table columns resizable by dragging #3582. resize columns with data bars

mnavasiolava committed:
save table interaction state #3685. save column visibility info

scottdraves committed:
Merge pull request #3694 from twosigma/tom/use-config-instead-of-pref

use beakerConfig instead of beakerConfigPref

EfimovVladimir committed:
create test for heatmap tutorial #3670

kpaharelau committed:
command N #3632

code improvement

mnavasiolava committed:
save table interaction state #3685. fix for saving alignment and type

mnavasiolava committed:
save table interaction state #3685. unify use of column order info

mnavasiolava committed:
save table interaction state #3685. col order fix

Commit:
#3658. disconnected dialogs still stack up

sshkalikov committed:
#3695. Cursor position is unintuitive when switching cells

Commit:
#3679. error when canceling calculations

sshkalikov committed:
#3423. Need word wrap in markdown mode

sshkalikov committed:
#3423. Need word wrap in markdown mode - and for code cells too

scottdraves committed:
Merge pull request #3678 from twosigma/cit/3667

create test for category plot notebook #3667

scottdraves committed:
Merge pull request #3661 from twosigma/marina/3659

add margin #3659

scottdraves committed:
Merge pull request #3662 from twosigma/marina/3660

add tooltips #3660

scottdraves committed:
Merge pull request #3682 from twosigma/vladimir/3674

add to help menu #3674

scottdraves committed:
Merge pull request #3684 from twosigma/cit/3683

Wrong chart size in Firefox browser #3683

scottdraves committed:
Merge pull request #3697 from twosigma/cit/3669

colors differ between plot and legend #3669

scottdraves committed:
Merge pull request #3701 from twosigma/sergey/3695

#3695. Cursor position is unintuitive when switching cells

scottdraves committed:
Merge pull request #3702 from twosigma/maxy/3679

#3679. error when canceling calculations

scottdraves committed:
Merge pull request #3703 from twosigma/sergey/3423

#3423. Need word wrap in markdown mode

scottdraves committed:
merge

scottdraves committed:
Merge pull request #3680 from twosigma/marina/3672

flatten column menu #3672

scottdraves committed:
merge

scottdraves committed:
Merge pull request #3681 from twosigma/marina/3673

tables: add hysteresis to column filter #3673

kpaharelau committed:
Merge remote-tracking branch 'remotes/origin/master' into alice/3632

kpaharelau committed:
Merge branch 'master' into alice/3632

Commit:
#3658. disconnected dialogs still stack up

kpaharelau committed:
command N #3632

code improvement

mnavasiolava committed:
save table interaction state #3685. fix for fixed row selection

mnavasiolava committed:
Merge branch 'master' into marina/3665

Conflicts:
	core/src/main/web/outputdisplay/bko-tabledisplay/bko-tabledisplay.js

mnavasiolava committed:
Merge branch 'master' into marina/3665

Conflicts:
	core/src/main/web/outputdisplay/bko-tabledisplay/bko-tabledisplay.js

EfimovVladimir committed:
test for heatmap-tutorial #3670

EfimovVladimir committed:
Merge branch 'master' of https://github.com/twosigma/beaker-notebook into vladimir/3670

Conflicts:
	test/protractorConf.js

kpaharelau committed:
Merge branch 'master' into alice/3532

EfimovVladimir committed:
Merge branch 'vladimir/3670'

mnavasiolava committed:
Merge branch 'master' into marina/3685

EfimovVladimir committed:
#3670

mnavasiolava committed:
save table interaction state #3685. fix for saving state after reorder columns, hiding one of them and alignment changes

EfimovVladimir committed:
merge #3670

EfimovVladimir committed:
check test #3705

EfimovVladimir committed:
check test #3670

sshkalikov committed:
#3706. bad column name should throw error

mnavasiolava committed:
save table interaction state #3685. save bars and heatmap

scottdraves committed:
Merge pull request #3668 from twosigma/marina/3665

ESC in search/filter box should be like clicking x #3665

scottdraves committed:
Merge pull request #3644 from twosigma/alice/3632

Replace cmd commands for new notebook commands

sshkalikov committed:
#3677. add tooltips to language manager

scottdraves committed:
Merge pull request #3664 from twosigma/marina/3663

Table width is broken after redraw #3663

EfimovVladimir committed:
#3670

scottdraves committed:
Merge pull request #3715 from twosigma/sergey/3706

#3706. bad column name should throw error

kpaharelau committed:
add menu to each value in beaker object interface #3532

still needs to focus on text field when the rename dialog opens, and fix color of the menu in default theme (should be green matching the language menu in cells in nonadvanced mode). -DONE

scottdraves committed:
Merge pull request #3716 from twosigma/sergey/3677

#3677. add tooltips to language manager

sshkalikov committed:
#3695. fix code style for previous change

scottdraves committed:
Merge pull request #3717 from twosigma/sergey/3695-1

#3695. fix code style for previous change

mnavasiolava committed:
make table columns resizable by dragging #3582. truncate cell text fix

scottdraves committed:
Merge pull request #3698 from twosigma/marina/3582-narrow-columns

Resizable columns. Truncate cell text. #3582

tomca32 committed:
#3676 - Separated value from label in language manager radio buttons and changed js language version format in bkr files

scottdraves committed:
Merge pull request #3700 from twosigma/maxy/3658

#3658. disconnected dialogs still stack up

tomca32 committed:
#3676 - Updated tutorials with new JS spec settings

scottdraves committed:
Merge pull request #3714 from twosigma/vladimir/3670

Vladimir/3670

tomca32 committed:
Refactored plugins showing and hiding spinner, eliminated code duplication accross plugins

tomca32 committed:
Refactored $.ajax to bkHelper.httpPost

tomca32 committed:
#3676 - Defaulting to default JS version if the version info is wrong in bkr file

scottdraves committed:
Merge pull request #3718 from twosigma/tom/3676

Separated value from label in language manager radio buttons and chan…

scottdraves committed:
Merge pull request #3722 from twosigma/tom/2972-refactor

Refactored plugins showing and hiding spinner

mnavasiolava committed:
save table interaction state #3685. save precision

kpaharelau committed:
add menu to each value in beaker object interface #3532

use different layout for beaker object and OutputContainer

mnavasiolava committed:
Merge branch 'master' into marina/3685

sshkalikov committed:
#3721. define language manager key

sshkalikov committed:
#3727. add to language manager tooltips

mnavasiolava committed:
save table interaction state #3685. table filter and search

mnavasiolava committed:
save table interaction state #3685. fix for column filter inputs overflow

sshkalikov committed:
#3696. Flicker when adding new cell - result section appears and is hidden

kpaharelau committed:
selection issues in code cells #1299

Go Doc Start on blur. Selection behaviour has been simplified

mnavasiolava committed:
save table interaction state #3685. column filter and search

mnavasiolava committed:
save table interaction state #3685. fix default double has 4 digits precision

EfimovVladimir committed:
Merge branch 'master' of https://github.com/twosigma/beaker-notebook into vladimir/3705

Conflicts:
	test/protractorConf.js

mnavasiolava committed:
save table interaction state #3685. save resize info

sshkalikov committed:
#3720. define reset kernel key cmd

kpaharelau committed:
better API to detect client OS #3501

API was added

sshkalikov committed:
#3720. define reset kernel key cmd

sshkalikov committed:
#3720. define reset kernel key cmd

kpaharelau committed:
better API to detect client OS #3501

Improve transformation beaker object model before output

sshkalikov committed:
#3709. add option for word wrap

sshkalikov committed:
#3728. A notebook is not marked as edited if cells are deleted or moved up/down

sshkalikov committed:
#3709. add option for word wrap - mark notebook edited on 'word wrap' option changing

sshkalikov committed:
#3721. define language manager key - move hotkey processing to mainapp.js

sshkalikov committed:
#3721. define language manager key - move hotkey processing to mainapp.js

sshkalikov committed:
#3720. define reset kernel key cmd - move hotkey processing to mainapp.js

sshkalikov committed:
#3720. define reset kernel key cmd - move hotkey processing to mainapp.js

sshkalikov committed:
#3720. define reset kernel key cmd - move hotkey processing to mainapp.js

EfimovVladimir committed:
fix sometimes all test failed #3705

scottdraves committed:
Merge pull request #3641 from twosigma/alice/3532

Menu in beaker object interface

scottdraves committed:
suffix, not prefix.  polish text, add more of them

sshkalikov committed:
#3720. add new changes to addevalplugins.js

scottdraves committed:
Merge pull request #3732 from twosigma/cit/1299

selection issues in code cells #1299

scottdraves committed:
Merge pull request #3734 from twosigma/cit/3501

better API to detect client OS #3501

scottdraves committed:
Merge pull request #3730 from twosigma/sergey/3727

#3727. add to language manager tooltips

scottdraves committed:
Merge pull request #3729 from twosigma/sergey/3721

#3721. define language manager key

scottdraves committed:
fix layout, issue #3739

scottdraves committed:
fix screenshot, issue #3692

scottdraves committed:
Update README.md

mnavasiolava committed:
precision key change #3724

kpaharelau committed:
add menu to each value in beaker object interface #3532

great. a few remaining details:
1) remove the "beaker object menu" tooltip
2) bring up to date with master
3) the triangle is wrong. it is narrower than the one in the language menu:
4) the text does not highlight and the cursor does not change to the finger in a large enough area. if you point between the triangle and the text, or below the triangle, then it does not activate. it should be active in the whole area around.

DONE

sshkalikov committed:
Merge remote-tracking branch 'origin/sergey/3720' into sergey/3720

EfimovVladimir committed:
fix sometimes all test failed #3705

kpaharelau committed:
output container hidden tabs not formatted correctly #3737

Fuctions 'safeWidth' and 'safeHeight' has been improved

mnavasiolava committed:
Revert "tables flash Processing... #3671. try to set timeout"

This reverts commit 772c49370bfea8b2cdee14b069555e9ca07ea3cf.

mnavasiolava committed:
tables flash "Processing..." #3671. add Processing indicator only if table has 500+ items

mnavasiolava committed:
Merge branch 'master' into marina/3671

Conflicts:
	core/src/main/web/outputdisplay/bko-tabledisplay/bko-tabledisplay.js

kpaharelau committed:
output container hidden tabs not formatted correctly #3737

use computed css styles

EfimovVladimir committed:
add tests for the SQL, Java, and Clojure tutorials #3719

mnavasiolava committed:
save table interaction state #3685. fix for table with index column

scottdraves committed:
update tutorial with new UI and document it

sshkalikov committed:
fix output of error during resetting python kernel ("ERROR:  A" bug)

sshkalikov committed:
fix output of error during resetting python kernel ("ERROR:  A" bug)

sshkalikov committed:
#3757. class "Foo" confuses Java??

sshkalikov committed:
Merge remote-tracking branch 'origin/sergey/3720' into sergey/3720

sshkalikov committed:
make 'reset' method of plugins return promise. after resetting plugin, wait until it reset

scottdraves committed:
Merge pull request #3733 from twosigma/marina/3685

save table interaction state #3685

scottdraves committed:
Merge pull request #3731 from twosigma/sergey/3696

#3696. Flicker when adding new cell - result section appears and is h…

scottdraves committed:
Merge pull request #3736 from twosigma/sergey/3709

#3709. add option for word wrap

scottdraves committed:
Merge pull request #3748 from twosigma/cit/3532

add menu to each value in beaker object interface #3532

scottdraves committed:
Merge pull request #3749 from twosigma/cit/3737

output container hidden tabs not formatted correctly #3737

scottdraves committed:
Merge pull request #3750 from twosigma/marina/3671

tables flash "Processing..." #3671

scottdraves committed:
Merge pull request #3751 from twosigma/sergey/3728

#3728. A notebook is not marked as edited if cells are deleted or mov…

mnavasiolava committed:
Merge branch 'master' into marina/3724

sshkalikov committed:
#3711. table text is too high

kpaharelau committed:
output container hidden tabs with table not formatted correctly for hidden cell #3753

I have fixed problem but found very strange issue.
Some time ago the code for a table redraw worked fine

 _.defer(function(){ scope.table.draw(false);});

 It doesn't work for unhided cell now.

 1. Open 'Output Containers and Layout Managers' tutorial
 2. Hide output cell for Stacked example
 3. Evaluate cell
 4. Unhide output cell

I found only one way to fix it

$timeout(function () {
              _.defer(function(){ scope.table.draw(false);});
              tableChanged = false;
            }, 0);

I ready for any new ideas

kpaharelau committed:
alt-f11 broken #3746

Problem was only for a markdown cells. Fixed

mnavasiolava committed:
table should not fill whole width #3758

scottdraves committed:
Merge pull request #3759 from twosigma/sergey/3757

#3757. class "Foo" confuses Java??

kpaharelau committed:
slash confuses java comment remover #3768

fixed. Parsing comments has been improved

EfimovVladimir committed:
fix test, issue #3739

scottdraves committed:
add architecture diagram so we can link to it from the wiki

scottdraves committed:
Merge pull request #3745 from twosigma/spot/3739

fix layout, issue #3739

sshkalikov committed:
#3760. java errors garbled

sshkalikov committed:
#3760. java errors garbled

EfimovVladimir committed:
changes of tests #3705

EfimovVladimir committed:
Merge branch 'master' of https://github.com/twosigma/beaker-notebook into vladimir/3705

Conflicts:
	test/tests/charting-tutorial.js

scottdraves committed:
Merge pull request #3769 from twosigma/marina/3758

table should not fill whole width #3758

scottdraves committed:
Merge pull request #3772 from twosigma/cit/3768

slash confuses java comment remover #3768

scottdraves committed:
Merge pull request #3766 from twosigma/cit/3746

alt-f11 broken #3746

scottdraves committed:
Merge pull request #3764 from twosigma/sergey/3711

#3711. table text is too high

scottdraves committed:
Merge pull request #3738 from twosigma/vladimir/3705

fix sometimes all test failed #3705

scottdraves committed:
improve status message

scottdraves committed:
Merge pull request #3763 from twosigma/marina/3724

precision key change #3724

sshkalikov committed:
#3762. java static import broken

mnavasiolava committed:
setting precision by kbd misses 1 column #3787

sshkalikov committed:
#3784. add "run all cells" shortcut

kpaharelau committed:
'SVGElement.offsetHeight' is deprecated warning after export to SVG #3755

Functions plotUtils.outerHeight and plotUtils.outerWidth has been added for replacing jquery fuctions outerWidth and outerHeight

mnavasiolava committed:
table search/filter should apply to CSV export #3779

mnavasiolava committed:
table search/filter should apply to CSV export #3779. fix for copy to clipboard

kpaharelau committed:
output container hidden tabs with table not formatted correctly for hidden cell #3753

My mistake with merge - FIXED

sshkalikov committed:
rename compiler, add copyright banner

sshkalikov committed:
rename compiler, add copyright banner

kpaharelau committed:
'SVGElement.offsetHeight' is deprecated warning after export to SVG #3755
fix problem for a hidden cell

sshkalikov committed:
fix some 'perform' methods, fix jumping on the init cell with error

EfimovVladimir committed:
Charting Tutorial tests are slow #3726

kpaharelau committed:
'SVGElement.offsetHeight' is deprecated warning after export to SVG #3755
fix another problem for a hidden cell

kpaharelau committed:
In some cases Run button doesn't work #3795

use setSelection function instead goDocStart

scottdraves committed:
Merge pull request #3735 from twosigma/sergey/3720

#3720. define reset kernel key cmd

tomca32 committed:
Removed Singleton annotation on abstract class ObservableChartSerializer

scottdraves committed:
Merge pull request #3765 from twosigma/cit/3753

output container hidden tabs with table not formatted correctly for h…

scottdraves committed:
Merge pull request #3797 from twosigma/tom/remove-singleton

Removed Singleton annotation on abstract class ObservableChartSerializer

EfimovVladimir committed:
Merge branch 'master' of https://github.com/twosigma/beaker-notebook into vladimir/3719

scottdraves committed:
Merge pull request #3789 from twosigma/sergey/3762

#3762. java static import broken

scottdraves committed:
Merge pull request #3791 from twosigma/sergey/3784

#3784. add "run all cells" shortcut

scottdraves committed:
Merge pull request #3790 from twosigma/marina/3787

setting precision by kbd misses 1 column #3787

scottdraves committed:
Merge pull request #3794 from twosigma/vladimir/3726

Charting Tutorial tests are slow #3726

EfimovVladimir committed:
Merge branch 'master' of https://github.com/twosigma/beaker-notebook into vladimir/3719

Conflicts:
	test/protractorConf.js

scottdraves committed:
Merge pull request #3792 from twosigma/cit/3755

'SVGElement.offsetHeight' is deprecated warning after export to SVG #…

scottdraves committed:
Merge pull request #3793 from twosigma/marina/3779

table search/filter should apply to CSV export #3779

EfimovVladimir committed:
test for SQL tutorial #3719

EfimovVladimir committed:
Merge branch 'master' of https://github.com/twosigma/beaker-notebook into vladimir/3719

mnavasiolava committed:
Menu gap in table header #3788

sshkalikov committed:
fix line numbers

sshkalikov committed:
Merge remote-tracking branch 'origin/sergey/3760' into sergey/3760

# Conflicts:
#	plugin/javash/src/main/java/com/twosigma/beaker/javash/utils/JavaEvaluator.java
#	plugin/javash/src/main/java/com/twosigma/beaker/javash/utils/JavaSourceCompiler.java
#	plugin/javash/src/main/java/org/abstractmeta/toolbox/compilation/compiler/impl/JavaSourceCompilerImpl.java

EfimovVladimir committed:
java tutorial test #3719

sshkalikov committed:
#3800. add reset backends to notebook menu

mnavasiolava committed:
clicking big run button moves cursor #3799

kpaharelau committed:
Plot's legend in exported SVG is broken #3752

problem was in the css - fixed

EfimovVladimir committed:
add test for Clojure tutorial #3719

EfimovVladimir committed:
fix clojure tutorial test #3719

kpaharelau committed:
Notebook: cell is moved up in case if changing cell id #3771

Sorting by rawIndex has been added

kpaharelau committed:
R spawns icon, loses focus #3355

DEFAULT_JVM_OPTS='"-Djava.awt.headless=true"' has been added to the start script

kpaharelau committed:
publish dialog focus should start in title #3767

focus=true has been added to the need cells

mnavasiolava committed:
plots should handle gmt timezones #3657

sshkalikov committed:
small empty-lines fixes

sshkalikov committed:
#3689. long line of code can bring up weird menu

EfimovVladimir committed:
expose cell IDs to testing framework #3774

EfimovVladimir committed:
use cell-id in SQL tutorial test

EfimovVladimir committed:
remake Charting tutorial test, use cell-id

sshkalikov committed:
#3777. tighter spacing of tables in advanced mode

sshkalikov committed:
#3777. move event name to GLOBALS

EfimovVladimir committed:
remake clojure, java tutorial tests, use cell id

scottdraves committed:
Merge pull request #3775 from twosigma/sergey/3760

#3760. java errors garbled

scottdraves committed:
Merge pull request #3802 from twosigma/marina/3788

Menu gap in table header #3788

scottdraves committed:
Merge pull request #3803 from twosigma/sergey/3800

#3800. add reset backends to notebook menu

scottdraves committed:
Merge pull request #3804 from twosigma/marina/3799

clicking big run button moves cursor #3799

scottdraves committed:
Merge pull request #3796 from twosigma/cit/3795

In some cases Run button doesn't work #3795

scottdraves committed:
Merge pull request #3805 from twosigma/cit/3752

Plot's legend in exported SVG is broken #3752

scottdraves committed:
Merge pull request #3813 from twosigma/cit/3767

publish dialog focus should start in title #3767

scottdraves committed:
Merge pull request #3810 from twosigma/cit/3771

Notebook: cell is moved up in case if changing cell id #3771

scottdraves committed:
Merge pull request #3812 from twosigma/cit/3355

R spawns icon, loses focus #3355

scottdraves committed:
Merge pull request #3814 from twosigma/marina/3657

plots should handle gmt timezones #3657

scottdraves committed:
Merge pull request #3819 from twosigma/sergey/3689

#3689. long line of code can bring up weird menu

scottdraves committed:
Merge pull request #3820 from twosigma/sergey/3777

#3777. tighter spacing of tables in advanced mode

EfimovVladimir committed:
EasyForm tests flaky #3725

EfimovVladimir committed:
Merge branch 'master' of https://github.com/twosigma/beaker-notebook into vladimir/3725

EfimovVladimir committed:
Heatmap tutorials are flaky #3770

EfimovVladimir committed:
add python tutorial test

EfimovVladimir committed:
Merge branch 'master' of https://github.com/twosigma/beaker-notebook into vladimir/3719

scottdraves committed:
Merge pull request #3816 from twosigma/vladimir/3774

expose cell IDs to testing framework #3774

scottdraves committed:
Merge pull request #3826 from twosigma/vladimir/3725

EasyForm tests flaky #3725

scottdraves committed:
Merge pull request #3835 from twosigma/vladimir/3770

Heatmap tutorials are flaky #3770

sshkalikov committed:
#3838. reset backends menu item

EfimovVladimir committed:
Merge branch 'vladimir/3774' of https://github.com/twosigma/beaker-notebook into vladimir/3719

Conflicts:
	test/tests/beaker.po.js

EfimovVladimir committed:
Merge branch 'master' of https://github.com/twosigma/beaker-notebook into vladimir/3719

EfimovVladimir committed:
add python tutorial test

EfimovVladimir committed:
fix test python tutorial

scottdraves committed:
Merge pull request #3847 from twosigma/sergey/3838

#3838. reset backends menu item

scottdraves committed:
Merge pull request #3806 from twosigma/vladimir/3719

Vladimir/3719

kpaharelau committed:
text cell stuck in TeX mode #3857

only text block scanned by markdown

scottdraves commented on core/src/main/web/app/helpers/helper.js:
```diff
@@ -547,13 +547,62 @@
+
+            var toDOM=function(str){
+              var d=document
+                ,i
```
style requires var on every tline.

kpaharelau committed:
text cell stuck in TeX mode #3857

I have used https://github.com/markdown-it/markdown-it instead https://github.com/chjj/marked.

kpaharelau committed:
text cell stuck in TeX mode #3857

>this works great but:
>1) if we are replacing markd, then replace it everywhere and remove it from bower.json & our src tree, and index_template.html
>2) exclude the lib subdir of markdown-it

DONE
